CRM/Contribute/BAO/Contribution.php

```diff
@@ -4188,30 +4190,29 @@ public static function updateMembershipBasedOnCompletionOfContribution($contribu
 
       // CRM-8141 update the membership type with the value recorded in log when membership created/renewed
       // this picks up membership type changes during renewals
-      // @todo this is almost certainly an obsolete sql call, the pre-change
-      // membership is accessible via $this->_relatedObjects
-      $sql = "
-SELECT    membership_type_id
-FROM      civicrm_membership_log
-WHERE     membership_id={$membershipParams['id']}
-ORDER BY  id DESC
-LIMIT 1;";
-      $dao = CRM_Core_DAO::executeQuery($sql);
-      if ($dao->fetch()) {
-        if (!empty($dao->membership_type_id)) {
-          $membershipParams['membership_type_id'] = $dao->membership_type_id;
-        }
+      $preChangeMembership = MembershipLog::get(FALSE)
+        ->addSelect('membership_type_id')
+        ->addWhere('membership_id', '=', 65)
```

Comment:
```suggestion
        ->addWhere('membership_id', '=', $membershipParams['id'])
```

Reply:
But isn't every membership ID 65? Oh, I see. It should be 42.